scripts/main.js

```diff
@@ -0,0 +1,558 @@
+                    }
+                }
+            });
+        }
+        if (isEng == true) {
+            keyBoard.querySelectorAll(".button").forEach(function (e) {
+                for (let j = 0; j < RU_KEYS_CAPS.length; j++) {
+                    if (EN_KEYS_CAPS[j] == e.getAttribute("data")) {
+                        e.querySelector("p").innerText = e.querySelector("p").innerText.toLowerCase();
+                    }
+                }
+                for (let j = 0; j < EN_KEYS_UP[0].length; j++) {
+                    if (EN_KEYS_UP[0][j].code == e.getAttribute("data")) {
+                        e.querySelector("p").innerText = EN_KEYS_UP[0][j].key;
+                    }
+                }
+            });
+        }
+    }
+    textArea.focus();
+    if (event.code == "Space") {
+        setSelectionRange(" ");
+    }
+    if (event.code == "Tab") {
+        setSelectionRange("\t");
+    }
+    if (event.code == "Enter") {
+        setSelectionRange("\n");
+    }
+    if (event.code == "Backspace") {
+        let start = textArea.selectionStart,
+            end = textArea.selectionEnd;
+        if (!textArea.setRangeText) { return; }
+        if (start >= end) {
+            if (start <= 0 || !textArea.setSelectionRange) { return; }
+            textArea.setSelectionRange(start - 1, start);
+        }
+        textArea.setRangeText("");
+        textArea.focus();
+    }
+    if (event.code == "Delete") {
+        let start = textArea.selectionStart,
+            end = textArea.selectionEnd;
+        if (!textArea.setRangeText) { return; }
+        if (start >= end) {
+            if (start <= 0 || !textArea.setSelectionRange) { return; }
+            textArea.setSelectionRange(start, start + 1);
+        }
+        textArea.setRangeText("");
+        textArea.focus();
+    }
+    if (event.code != "Backspace" && event.code != "Enter" && event.code != "Tab" && event.code != "Space" && event.code != "Delete" && event.code != "CapsLock") {
+        document.querySelector(`.button[data="${event.keyCode}"]`) !== null ? setSelectionRange(document.querySelector(`.button[data="${event.keyCode}"]`).querySelector("p").innerText) : null;
+    }
+    document.querySelector(`.button[data="${event.keyCode}"]`) === null ? document.querySelector(`.button[data="${event.code}"]`).classList.add("active") : document.querySelector(`.button[data="${event.keyCode}"]`).classList.add("active");
+});
+
+function setSelectionRange(text) {
+    textArea.focus();
+    textArea.setRangeText(text, textArea.selectionStart, textArea.selectionEnd, "end");
+
+}
+
+document.body.addEventListener("keyup", function (event) {
+    keyBoard.querySelectorAll(".button").forEach(function (e) {
+        e.classList.remove("active");
+    });
+    document.querySelector(`.button[data="${event.code}"]`) === null ? null : document.querySelector(`.button[data="${event.code}"]`).classList.remove("active-shift");
```

Comment:
if null => null ?)
`if (document.querySelector(`.button[data="${event.code}"]`)) {document.querySelector(`.button[data="${event.code}"]`).classList.remove("active-shift");}`

Reply:
> if null => null ?)
> `if (document.querySelector(`.button[data="${event.code}"]`)) {document.querySelector(`.button[data="${event.code}"]`).classList.remove("active-shift");}`

Thank you for your comment. I agree that this is also very stupid.